libs/zzz/page-optimize-pando/src/Optimize/index.tsx

```diff
@@ -0,0 +1,241 @@
+import { Solver } from '@genshin-optimizer/zzz/solver-pando'
+import { getCharStat, getWengineStat } from '@genshin-optimizer/zzz/stats'
+import { WorkerSelector } from '@genshin-optimizer/zzz/ui'
+import CloseIcon from '@mui/icons-material/Close'
+import TrendingUpIcon from '@mui/icons-material/TrendingUp'
+import {
+  Box,
+  Button,
+  CardContent,
+  CardHeader,
+  Divider,
+  LinearProgress,
+  Typography,
+} from '@mui/material'
+import {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+import { useTranslation } from 'react-i18next'
+import GeneratedBuildsDisplay from './GeneratedBuildsDisplay'
+
+export default function Optimize() {
+  const { key: characterKey } = useCharacterContext()!
+  const { optConfigId } = useCharOpt(characterKey)!
+  const { database } = useDatabaseContext()
+  useEffect(() => {
+    if (optConfigId) return
+    const newOptConfigId = database.optConfigs.new()
+    database.charOpts.set(characterKey, {
+      optConfigId: newOptConfigId,
+    })
+  }, [database, optConfigId, characterKey])
+  if (!optConfigId) return null
+  return (
+    <OptConfigProvider optConfigId={optConfigId}>
+      <OptimizeWrapper />
+      <GeneratedBuildsDisplay />
+    </OptConfigProvider>
+  )
+}
+
+function OptimizeWrapper() {
+  const { t } = useTranslation('optimize')
+  const { database } = useDatabaseContext()
+  const calc = useZzzCalcContext()
+  const { key: characterKey } = useCharacterContext()!
+  const { target } = useCharOpt(characterKey)!
+  const [numWorkers, setNumWorkers] = useState(8)
+  const [progress, setProgress] = useState<ProgressResult | undefined>(
+    undefined
+  )
+  const { optConfig, optConfigId } = useContext(OptConfigContext)
+  const discsBySlot = useMemo(
+    () =>
+      database.discs.values.reduce(
+        (discsBySlot, disc) => {
+          discsBySlot[disc.slotKey].push(disc)
+          return discsBySlot
+        },
+        {
+          1: [],
+          2: [],
+          3: [],
+          4: [],
+          5: [],
+          6: [],
+        } as Record<DiscSlotKey, ICachedDisc[]>
+      ),
+    [database.discs.values]
+  )
+  const wengines = useMemo(() => {
+    const { specialty } = getCharStat(characterKey)
+    return database.wengines.values.filter(({ key }) => {
+      // filter by path
+      const { type } = getWengineStat(key)
+      return specialty === type
+    })
+  }, [characterKey, database.wengines.values])
+  const totalPermutations = useMemo(
+    () =>
+      Object.values(discsBySlot).reduce(
+        (total, discs) => total * discs.length,
+        1
+      ) * wengines.length,
+    [wengines.length, discsBySlot]
+  )
+
+  const [optimizing, setOptimizing] = useState(false)
+
+  // Provides a function to cancel the work
+  const cancelToken = useRef(() => {})
+  //terminate worker when component unmounts
+  useEffect(() => () => cancelToken.current(), [])
+
+  const onOptimize = useCallback(async () => {
```

Comment:
solver calculates and produces 0 results for me

Reply:
It is generating results for me, I think might be something missing in your setup (will likely narrow this down in the future)
![image](https://github.com/user-attachments/assets/e547a8fe-544a-422b-aa65-b1cd6726fcea)

Reply:
forgot to add a wengine to my db